src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -0,0 +1,945 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// A source method that can have attributes, including a member method, accessor, or local function.
+    /// </summary>
+    internal abstract class SourceMethodSymbolWithAttributes : SourceMethodSymbol, IAttributeTargetSymbol
+    {
+        private CustomAttributesBag<CSharpAttributeData> _lazyCustomAttributesBag;
+        private CustomAttributesBag<CSharpAttributeData> _lazyReturnTypeCustomAttributesBag;
+
+        // some symbols may not have a syntax (e.g. lambdas, synthesized event accessors)
+        protected readonly SyntaxReference syntaxReferenceOpt;
+        protected SourceMethodSymbolWithAttributes(SyntaxReference syntaxReferenceOpt)
+        {
+            this.syntaxReferenceOpt = syntaxReferenceOpt;
+        }
+
+#nullable enable
+        internal CSharpSyntaxNode? GetBinderContextNode()
+        {
+            switch (SyntaxNode)
```

Comment:
>switch (SyntaxNode) [](start = 12, length = 19)

Would it make sense to move ```internal (BlockSyntax blockBody, ArrowExpressionClauseSyntax arrowBody) Bodies``` into this class and use it to calculate result of this function?

Reply:
No, because it will miss some cases related to constructor initializers.

---
In reply to: [354029077](https://github.com/dotnet/roslyn/pull/40011#discussion_r354029077) [](ancestors = 354029077)